notebook.ipynb

```diff
@@ -0,0 +1,365 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile components/write_to_hub_controlnet/fondant_component.yaml\n",
+    "name: Write to hub\n",
+    "description: Component that writes a dataset to the hub\n",
+    "image: fndnt/write_to_hf_hub:0.6.2\n",
+    "\n",
+    "consumes:\n",
+    "  images:\n",
+    "    fields:\n",
+    "      data:\n",
+    "        type: binary\n",
+    "\n",
+    "  captions:\n",
+    "    fields:\n",
+    "      text:\n",
+    "        type: string\n",
+    "\n",
+    "  segmentations:\n",
+    "    fields:\n",
+    "      data:\n",
+    "        type: binary\n",
+    "\n",
+    "args:\n",
+    "  hf_token:\n",
+    "    description: The hugging face token used to write to the hub\n",
+    "    type: str\n",
+    "  username:\n",
+    "    description: The username under which to upload the dataset\n",
+    "    type: str\n",
+    "  dataset_name:\n",
+    "    description: The name of the dataset to upload\n",
+    "    type: str\n",
+    "  image_column_names:\n",
+    "    description: A list containing the image column names. Used to format to image to HF hub format\n",
+    "    type: list\n",
+    "    default: []\n",
+    "  column_name_mapping:\n",
+    "    description: Mapping of the consumed fondant column names to the written hub column names\n",
+    "    type: dict\n",
+    "    default: {}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "USERNAME = \n",
+    "HF_TOKEN = \n",
+    "\n",
+    "write_to_hub_controlnet = ComponentOp(\n",
+    "    component_dir=\"components/write_to_hub_controlnet\",\n",
+    "    arguments={\n",
+    "        \"username\": USERNAME ,\n",
+    "        \"hf_token\": HF_TOKEN ,\n",
+    "        \"dataset_name\": \"controlnet-interior-design\",\n",
+    "        \"image_column_names\": [\"images_data\"],\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipeline.add_op(write_to_hub_controlnet, dependencies=segment_images_op)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Execute the pipeline\n",
+    "\n",
+    "The pipeline will generate the prompts, retreive matching images in the laion dataset and download then and finally will generate corresponding captions and segmentations needed before writing the dataset to the HF hub.\n",
+    "\n",
+    "We can execute our pipeline. Fondant provides various executors, and in this case, we are using the `DockerRunner` for local execution, which utilizes docker-compose under the hood."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "compiler = DockerCompiler()\n",
```

Comment:
Can we add a cell which sets the environment variable to pull the linux/amd64 images. I've added it for consistency to the RAG notebook. 

```
import os
os.environ["DOCKER_DEFAULT_PLATFORM"]="linux/amd64"
```

Reply:
This should probably depend on the OS the user is running, right?

Reply:
Nvm, it should always be this.